### PR TITLE
Civ 3178 b

### PR DESCRIPTION
**Pull request by cj1993**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java [hunk 1]
```diff
@@ -694,9 +688,110 @@ private String getBody(CaseData caseData) {
...
+        updatedData.disposalHearingCostsToggle(checkList);
+        updatedData.disposalHearingApplicationsOrderToggle(checkList);
+        updatedData.smallClaimsHearingToggle(checkList);
+        updatedData.smallClaimsMethodToggle(checkList);
+        updatedData.smallClaimsDocumentsToggle(checkList);
+        updatedData.smallClaimsWitnessStatementToggle(checkList);
+        updatedData.disposalHearingApplicationsOrderToggle(checkList);
+    }
+
+    private List<String> setPreferredLocationFirst(List<LocationRefData> locations, CaseData caseData) {
```
Comment (akriti-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java:722]: is there a requirement to have the preferred location first in the list?

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java [hunk 2]
```diff
@@ -259,7 +365,7 @@ void shouldPrePopulateOrderDetailsPages() {
 
             assertThat(response.getData()).extracting("disposalHearingBundle").extracting("input")
                 .isEqualTo("At least 7 days before the disposal hearing, "
-                                + "the claimant must upload to the Digital Portal");
+                               + "the claimant must upload to the Digital Portal");
 
             assertThat(response.getData()).extracting("disposalHearingNotes").extracting("input")
                 .isEqualTo("This Order has been made without a hearing. Each party has the right to apply to have"
```
Comment (akriti-hmcts) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java:371]: Please add unit test for preferred court to appear first.

**Review by akriti-hmcts — COMMENTED** [review]
added 2 minor comments

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java [hunk 3]
```diff
@@ -694,9 +688,110 @@ private String getBody(CaseData caseData) {
...
+    }
 
-        return locationRefDataService.getCourtLocations(authToken);
+    private List<String> getLocationsFromList(final List<LocationRefData> locations) {
+        return locations.stream()
+            .map(location -> location.getSiteName()
+                + " - " + location.getCourtAddress()
+                + " - " + location.getPostcode())
+            .collect(Collectors.toList());
     }
```
Comment (akriti-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java:796]: case management location logic needs to be added in this class as per -

https://tools.hmcts.net/confluence/display/CRef/Civil+Work+Allocation+Configuration#CivilWorkAllocationConfiguration-2.2.4CaseManagementLocationrules

**Review by akriti-hmcts — CHANGES_REQUESTED** [review]
add case management location logic in handler

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java (hunk 1 above)
Comment (cj1993) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java:722]: Yes

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java (hunk 2 above)
Comment (cj1993) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java:385]: Done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java [hunk 4]
```diff
@@ -128,35 +139,18 @@ private CallbackResponse prePopulateOrderDetailsPages(CallbackParams callbackPar
...
-        updatedData.smallClaimsWitnessStatementToggle(checkList);
+        updatedData.disposalHearingMethodInPerson(locationsList);
+        updatedData.fastTrackMethodInPerson(locationsList);
+        updatedData.smallClaimsMethodInPerson(locationsList);
+
+        List<OrderDetailsPagesSectionsToggle> checkList = List.of(OrderDetailsPagesSectionsToggle.SHOW);
+        setCheckList(updatedData, checkList);
 
         DisposalHearingJudgesRecital tempDisposalHearingJudgesRecital = DisposalHearingJudgesRecital.builder()
             .input("Upon considering the claim form, particulars of claim, statements of case"
```
Comment (sampankumar) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java:156]: is it possible to make this text in a Constant please

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java [hunk 5]
```diff
@@ -694,9 +688,112 @@ private String getBody(CaseData caseData) {
...
+        updatedData.disposalHearingCostsToggle(checkList);
+        updatedData.disposalHearingApplicationsOrderToggle(checkList);
+        updatedData.smallClaimsHearingToggle(checkList);
+        updatedData.smallClaimsMethodToggle(checkList);
+        updatedData.smallClaimsDocumentsToggle(checkList);
+        updatedData.smallClaimsWitnessStatementToggle(checkList);
+        updatedData.disposalHearingApplicationsOrderToggle(checkList);
+    }
+
+    private List<String> setPreferredLocationFirst(List<LocationRefData> locations, CaseData caseData) {
```
Comment (akriti-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java:722]: this method is too long and not easy to read, Ideally the case management logic should go in a separate method. I would leave it for now but we will have to refactor it at some point.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java [hunk 6]
```diff
@@ -14,15 +14,19 @@
 import uk.gov.hmcts.reform.civil.callback.CallbackParams;
```
Comment (akriti-hmcts) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandlerTest.java:14]: No unit test for case management location. Can we please add that

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java [hunk 7]
```diff
@@ -694,9 +688,112 @@ private String getBody(CaseData caseData) {
...
+            locationLabel = caseData.getCourtLocation().getApplicantPreferredCourt();
+        }
+
+        var preferredLocation =
+            locations
+                .stream()
+                .filter(locationRefData -> checkLocation(locationRefData,
+                                                         locationLabel)).findFirst();
+
+        if (caseData.getOrderType() == OrderType.DECIDE_DAMAGES && preferredLocation.isPresent()) {
```
Comment (akriti-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateSDOCallbackHandler.java:743]: is there a reason we are not checking the following -
!(SPEC_CLAIM.equals(caseData.getSuperClaimType())
and using OrderType instead?

**Review by akriti-hmcts — CHANGES_REQUESTED** [review]

**Review by sampankumar — APPROVED** [review]

**Pull request closed without merging**